### PR TITLE
add service user Redis ACL fields

### PR DESCRIPTION
**Pull request by ivan-savciuc**

### CODE REVIEW

File: service_user.go [hunk 1]
```diff
@@ -27,14 +31,20 @@ type (
 	// CreateServiceUserRequest are the parameters required to create a
 	// ServiceUser.
 	CreateServiceUserRequest struct {
-		Username string `json:"username"`
+		Username           string   `json:"username"`
+		RedisACLCategories []string `json:"redis_acl_categories,omitempty"`
```
Comment (rikonen) [inline service_user.go:35]: Doesn't look like these keys are placed correctly. They should be under `access_control`. Also when the `access_control` key is present it must contain all three lists.

File: service_user.go [hunk 2]
```diff
@@ -27,14 +31,20 @@ type (
...
 	ModifyServiceUserRequest struct {
-		Operation      *string `json:"operation"`
-		Authentication *string `json:"authentication,omitempty"`
-		NewPassword    string  `json:"new_password"`
+		Operation          *string  `json:"operation"`
+		Authentication     *string  `json:"authentication,omitempty"`
+		NewPassword        string   `json:"new_password"`
+		RedisACLCategories []string `json:"redis_acl_categories,omitempty"`
+		RedisACLCommands   []string `json:"redis_acl_commands,omitempty"`
+		RedisACLKeys       []string `json:"redis_acl_keys,omitempty"`
```
Comment (rikonen) [inline service_user.go:47]: As far as I can see updating these isn't currently possible. Probably it should be.

File: service_user.go [hunk 3]
```diff
@@ -11,11 +11,15 @@ import (
...
-		AccessKey  string `json:"access_key"`
+		Username                    string   `json:"username"`
+		Password                    string   `json:"password"`
+		Type                        string   `json:"type"`
+		AccessCert                  string   `json:"access_cert"`
+		AccessKey                   string   `json:"access_key"`
+		AccessCertNotValidAfterTime string   `json:"access_cert_not_valid_after_time"`
+		RedisACLCategories          []string `json:"redis_acl_categories"`
+		RedisACLCommands            []string `json:"redis_acl_commands"`
+		RedisACLKeys                []string `json:"redis_acl_keys"`
```
Comment (rikonen) [inline service_user.go:22]: These are also not directly under user but in access_control object.

**Review by rikonen — CHANGES_REQUESTED** [review]

File: service_user.go [hunk 4]
```diff
@@ -11,11 +11,19 @@ import (
...
-		Type       string `json:"type"`
-		AccessCert string `json:"access_cert"`
-		AccessKey  string `json:"access_key"`
+		Username                    string         `json:"username"`
+		Password                    string         `json:"password"`
+		Type                        string         `json:"type"`
+		AccessCert                  string         `json:"access_cert"`
+		AccessKey                   string         `json:"access_key"`
+		AccessCertNotValidAfterTime string         `json:"access_cert_not_valid_after_time"`
+		ServiceControl              ServiceControl `json:"service_control"`
```
Comment (rikonen) [inline service_user.go:20]: access_control, not service_control

**Review by rikonen — DISMISSED** [review]

File: service_user.go (hunk 4 above)
Comment (rikonen) [reply service_user.go:20]: Also this needs to be omitted except when it's being set for Redis.

Comment (ivan-savciuc) [reply service_user.go]: Ups sorry did change the name of the field

**Review by rikonen — APPROVED** [review]

**Pull request merged**